### PR TITLE
Add barcode lookup route

### LINKED ISSUE #62: Lookup by Barcode
Looking up an item by barcode is a use-case for the ReCAP middleware project. Will support the ReCAP accession process for newly acquired materials.

### PR DESCRIPTION
**Pull request by mzelesky**
Closes #62.

### CODE REVIEW

**Comment by mzelesky** [comment]
There is a bug with the trim_punctuation feature in the latest version of traject (2.3.2). For now, we have rolled our version of traject back to 2.3.1.

File: app/controllers/concerns/formatting_concern.rb [hunk 1]
```diff
@@ -36,6 +36,10 @@ def pass_records_through_xml_parser(records)
     end
   end
 
+  def valid_barcode(barcode='')
+    barcode =~ /^(32101[0-9]{9}|PULTST[0-9]{5})$/
```
Comment (tampakis) [inline app/controllers/concerns/formatting_concern.rb]: Could the method be renamed `valid_barcode?`.
Also the method should either return true or false, not 0 or nil. Try this to get a boolean response:
```
(barcode =~ /^(32101[0-9]{9}|PULTST[0-9]{5})$/) == 0
```

**Comment by mzelesky** [comment]
Ok, I have made that change.

**Comment by tampakis** [comment]
Cool

**Pull request merged**